lib/gitsh/git_command_list.rb

```diff
@@ -0,0 +1,56 @@
+module Gitsh
+  class GitCommandList
+    def initialize(env)
+      @env = env
+    end
+
+    def to_a
+      try_using(commands_from_list_cmds) do
+        try_using(commands_from_help('help -a --no-verbose')) do
+          try_using(commands_from_help('help -a'))
+        end
+      end
+    end
+
+    private
+
+    attr_accessor :env
+
+    def try_using(result, default = [])
```

Comment:
What do you think about using a defaulted named parameter here? That way if a default is provided, it’s a little more self-documenting.